src/main/java/org/broadinstitute/hellbender/tools/exome/BayesianHetPulldownCalculator.java

```diff
@@ -483,29 +518,36 @@ else if (base == alleleAlt) {
     }
 
     /**
-     * Infer the alt allele base from the pileup. We choose the base with the highest frequency
-     * that is not Ref (this coincides with the maximum likelihood estimation).
+     * Infer the alt allele base from the pileup: we pick the base with the highest frequency that is not Ref as Alt.
+     * This is the maximum likelihood estimation. <br>
      *
-     * * Remark: while this is definitely not the perfect way to do it, it has no serious pitfalls:
+     * <b>Remark</b>: While this is definitely not the perfect way to do it and it is desirable to have both Ref/Alt
+     * allele information, it has no serious pitfalls either:<BR><BR>
+     * <ul>
+     *     <li>
+     *         If the reads are balanced between between Ref and Alt, it correctly infers the Alt base.
+     *     </li>
+     *     <li>
+     *         If the input is all Ref, then Alt is chosen randomly, but it's fine since there are no Alt counts in
+     *         the pileup anyway and the likelihoods are not affected. If the input is all Alt, it still works properly.
+     *     </li>
+     *     <li>
+     *         If the input in all Ref + a few read/mapping errors, then Alt will be chosen as the most frequent
+     *         erroneous base; again, this is fine because Hom likelihood is far greater than Het likelihood in this
+     *         case anyway. Likewise, if the input is all Alt + a few errors, it still works properly.
+     *     </li>
+     * </ul>
      *
-     * - if the input is balanced between betweet ref and alt, it correctly chooses alt.
-     * - if the input is all ref, then alt is chosen randomly, but it's OK since there are no alt counts
-     *   in the pileup anyway.
-     * - if the input is all alt, we are good.
-     * - if the input in all ref + a few errors, then alt will be choosen as the most frequent erroneous
-     *   base; again, this is OK because Hom likelihood is far greater than Het likelihood in this case
-     * - if the input is all alt + a few errors, we are good.
-
      * @param baseQualities map from bases to error probabilities
      * @param refBase the ref allele base
      * @return the likely alt allele
      */
     @VisibleForTesting
-    public static Nucleotide inferAltFromPileup(final Map<Nucleotide, ArrayList<BaseQuality>> baseQualities,
+    public static Nucleotide inferAltFromPileup(final Map<Nucleotide, List<BaseQuality>> baseQualities,
                                                 final Nucleotide refBase) {
 
         /* sort the bases in the descending order by their frequency */
-        Nucleotide[] bases = {Nucleotide.A, Nucleotide.C, Nucleotide.G, Nucleotide.T};
+        Nucleotide[] bases = BASES;
```

Comment:
I know that we are single-thread, yet as a matter of principle you must not modify the order of an array that is declared as a constant: ```final Nucleotide[] bases = BASE.clone()```